docarray/index/backends/elastic.py

```diff
@@ -90,6 +94,13 @@ def __init__(self, db_config=None, **kwargs):
         mappings.update(self._db_config.index_mappings)
 
         for col_name, col in self._column_infos.items():
+            if issubclass(col.docarray_type, AnyDocArray):
+                sub_db_config = copy.deepcopy(self._db_config)
+                sub_db_config.index_name = f'{self._index_name}__{col_name}'
+                self._subindices[col_name] = self.__class__[col.docarray_type.doc_type](
+                    db_config=sub_db_config, subindex=True
+                )
+                continue
```

Comment:
I think this same code is repeated in all backends. Can't it be done in the abstract base class?

I think in v1 we managed to do this sort of thing in the base class, any only exposed the `index_name` as something that the backend implementation needs to tackle, while the rest was "automated". I think the same should be possible here

Reply:
This implies that `index_name` is now an alias that can be used with any backend, right?
If so, this needs to be documented. In parts of the docs I tell people to look at the fields of `DBConfig` to see all available configs; if they now see `collection_name` and `index_name` that both do the same it will be confusing